airflow/dag_processing/manager.py

```diff
@@ -765,9 +765,13 @@ def _refresh_dag_dir(self):
                 else:
                     dag_filelocs.append(fileloc)
 
+            processor_subdir = None
+            if self.standalone_dag_processor:
+                processor_subdir = self.get_dag_directory()
+
             SerializedDagModel.remove_deleted_dags(
                 alive_dag_filelocs=dag_filelocs,
-                processor_subdir=self.get_dag_directory(),
+                processor_subdir=processor_subdir,
```

Comment:
Can you add tests to prove this?

Reply:
While that worked, I was misunderstanding the `remove_deleted_dags` method and found out it was a bug in that method. See latest commit